### PR TITLE
[DOCS] Move `store` utils to individual reference pages

### LINKED ISSUE #467: Add store utilities into their own pages in Reference
Separate / add store utilities to a section in the reference similar to what is done here:

https://www.solidjs.com/docs/latest/api#store-utilities

### PR DESCRIPTION
**Pull request by dyyyl**
This PR separates the utilities documented in [Store Utilities](https://docs.solidjs.com/reference/stores/store-utilities) into individual pages for each utility.

closes #467

### CODE REVIEW

File: src/routes/reference/store-utilities/modify-mutable.mdx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+```tsx
+function modifyMutable<T>(mutable: T, modifier: (state: T) => T): void;
+```
+
+This helper function simplifies making multiple changes to a mutable Store (as returned by [createMutable](#createMutable)) in a single [batch](/reference/reactive-utilities/batch), so that dependant computations update just once instead of once per update.
+The first argument is the mutable Store to modify, and the second argument is a Store modifier such as those returned by [reconcile]('#reconcile') or [produce](#produce).
+
+<Callout type="caution">
+	If you pass in your own modifier function, beware that its argument is an
+	unwrapped version of the Store.
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/modify-mutable.mdx:14]:
```suggestion
	When passing in your own modifier function, it's important to be aware that its argument is an unwrapped version of the store.
```

File: src/routes/reference/store-utilities/modify-mutable.mdx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+</Callout>
+
+For example, suppose we have a UI depending on multiple fields of a mutable:
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/modify-mutable.mdx:17]:
```suggestion
For example, if the UI depends on multiple fields of a mutable:
```

File: src/routes/reference/store-utilities/modify-mutable.mdx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+Modifying n fields in sequence will cause the UI to update n times:
+
+```tsx
+state.user.firstName = "Jane";
+state.user.lastName = "Doe";
+```
+
+To trigger just a single update, we could modify the fields in a `batch`:
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/modify-mutable.mdx:39]:
```suggestion
To trigger just a single update, the fields can be modified using a `batch`:
```

File: src/routes/reference/store-utilities/modify-mutable.mdx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+```tsx
+import { batch } from "solid-js";
+
+batch(() => {
+	state.user.firstName = "Jane";
+	state.user.lastName = "Doe";
+});
+```
+
+`modifyMutable` combined with reconcile or produce provides two alternate ways to do similar things:
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/modify-mutable.mdx:50]:
```suggestion
`modifyMutable` combined with `reconcile` or `produce` provides two alternate ways to do similar things:
```

File: src/routes/reference/store-utilities/produce.mdx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+```ts
+function produce<T>(
+	fn: (state: T) => void
+): (
+	state: T extends NotWrappable ? T : Store<T>
+) => T extends NotWrappable ? T : Store<T>;
+```
+
+An [Immer](https://immerjs.github.io/immer/) inspired API for Solid's Store objects that allows for localized mutation.
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/produce.mdx:13]:
```suggestion
`produce` is an [Immer](https://immerjs.github.io/immer/) inspired API for Solid's Store objects that allow for localized mutation.
```

File: src/routes/reference/store-utilities/reconcile.mdx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+function reconcile<T>(
+	value: T | Store<T>,
+	options?: {
+		key?: string | null;
+		merge?: boolean;
+	} = { key: "id" }
+): (
+	state: T extends NotWrappable ? T : Store<T>
+) => T extends NotWrappable ? T : Store<T>;
+```
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/reconcile.mdx:15]: similar to other comments - please move below first blurb

File: src/routes/reference/store-utilities/reconcile.mdx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+Diffs data changes when we can't apply granular updates.
+Useful for when dealing with immutable data from stores or large API responses.
+
+The key option is used when available to match items.
+By default the `merge` option false does referential checks where possible to determine equality and replaces where items are not referentially equal.
+`merge` as `true` pushes all diffing to the leaves and effectively morphs the previous data to the new value.
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/reconcile.mdx:22]:
```suggestion
`reconcile` is designed for diffing data changes in situations where granular updates cannot be applied. 
This is useful when dealing with immutable data from stores or handling large API responses.
`reconcile` has a key option that can be used when available to match items.

`value` accepts either a value of type `T` or a Store containing values of type `T`. 
This represents the data to be reconciled.

      The `reconcile` function helps manage data changes by performing a diffing process, making it particularly handy in scenarios where applying granular updates is challenging or inefficient. 
      The `key` and `merge` options provide flexibility to customize the reconciliation process based on specific needs.
```

File: src/routes/reference/store-utilities/reconcile.mdx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+onCleanup(() => unsubscribe());
+
+```
+
+##### Options
+
+| Option | Type    | Default | Description                        |
+| ------ | ------- | ------- | ---------------------------------- |
+| key    | string  | "id"    | The key to use for matching items. |
+| merge  | boolean | false   | Whether to merge or replace items. |
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/reconcile.mdx:38]:
```suggestion
| Option | Type    | Default | Description                        |
| ------ | ------- | ------- | ---------------------------------- |
| key    | string  | "id"    |  Specifies the key to be used for matching items during reconciliation |
| merge  | boolean | false   | When merge is false, referential checks are performed where possible to determine equality, and items that are not referentially equal are replaced. When merge is true, all diffing is pushed to the leaves, effectively morphing the previous data to the new value. |
```

File: src/routes/reference/store-utilities/unwrap.mdx [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+title: unwrap
+---
+
+```tsx
+import { unwrap } from "solid-js/store";
+
+function unwrap(store: Store<T>): T;
+```
+
+Returns the underlying data in the store without a proxy.
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/unwrap.mdx:11]:
```suggestion
`unwrap` returns the underlying data in the store without a proxy.
```

File: src/routes/reference/store-utilities/unwrap.mdx [hunk 10]
```diff
@@ -0,0 +1,11 @@
+---
+title: unwrap
+---
+
+```tsx
+import { unwrap } from "solid-js/store";
+
+function unwrap(store: Store<T>): T;
+```
```
Comment (LadyBluenotes) [inline src/routes/reference/store-utilities/unwrap.mdx:9]: same comment as other pages

**Review by LadyBluenotes — COMMENTED** [review]
Thank you! Just a few changes before we're good to merge :)

**Comment by LadyBluenotes** [comment]
@dyyyl have all the above been resolved with the updated commits?

**Comment by dyyyl** [comment]
@LadyBluenotes anything here marked as resolved have been auto-resolved via committing the suggestions made in review. I have not yet had time to resolve anything unmarked.

**Comment by LadyBluenotes** [comment]
@danieljcafonso / @atilafassina can either of you read these over to see if any other changes need to be added / made?

**Pull request closed without merging**